core/src/software/aws/toolkits/core/telemetry/TelemetryBatcher.kt

```diff
@@ -76,6 +81,13 @@ open class DefaultTelemetryBatcher(
         flush(retry, isTelemetryEnabled.get())
     }
 
+    @Synchronized
+    override fun shutdownAndFlush(batcher: TelemetryBatcher) {
```

Comment:
nit: this method name is weird, I first read it as we are shutting down and flushing the batcher that is passed in. Maybe something like `setBatcher` would be better? What we are actually doing is setting a new batcher in a way that works.